sdk/core/azure-core/test/ut/retry_policy_test.cpp

```diff
@@ -55,25 +55,31 @@ class RetryPolicyTest final : public RetryPolicy {
           int32_t,
           std::chrono::milliseconds&,
           double)> shouldRetryOnResponse)
-      : RetryPolicy(retryOptions),
-      m_shouldRetryOnTransportFailure(
-          shouldRetryOnTransportFailure != nullptr
-            ? shouldRetryOnTransportFailure
-            : [&](auto options, auto attempt, auto retryAfter, auto jitter) {
+      : RetryPolicy(retryOptions), m_shouldRetryOnTransportFailure(shouldRetryOnTransportFailure),
+        m_shouldRetryOnResponse(shouldRetryOnResponse)
+  {
+    if (m_shouldRetryOnTransportFailure == nullptr)
+    {
+      m_shouldRetryOnTransportFailure
+          = [&](auto options, auto attempt, auto retryAfter, auto jitter) {
               retryAfter = std::chrono::milliseconds(0);
               auto ignore = decltype(retryAfter)();
               return RetryPolicy::ShouldRetryOnTransportFailure(options, attempt, ignore, jitter);
-        }),
-        m_shouldRetryOnResponse(
-            shouldRetryOnResponse != nullptr
-            ? shouldRetryOnResponse
-            : [&](RawResponse const& response, auto options, auto attempt, auto retryAfter, auto jitter) {
-              retryAfter = std::chrono::milliseconds(0);
-              auto ignore = decltype(retryAfter)();
-              return RetryPolicy::ShouldRetryOnResponse(
-                  response, options, attempt, ignore, jitter);
-        })
-  {
+            };
+    }
+
+    if (m_shouldRetryOnResponse == nullptr)
```

Comment:
Just a bit question, could this be written as follows?
```C++
if (!m_shouldRetryOnResponse)
```

Reply:
For sure, it is a matter of taste. I prefer `==/!= nullptr`, for non-booleans, and `!isSomething` for booleans.

Reply:
Yeah, seems reasonable. I think I have seen both in the codebase.

Reply:
I think, especially here, when the callback is totally named, how a boolean could've been named: `shouldSomething`